controllers/openstackdataplanerole_controller.go

```diff
@@ -210,8 +210,14 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
 	// all setup tasks complete, mark SetupReadyCondition True
 	instance.Status.Conditions.MarkTrue(dataplanev1beta1.SetupReadyCondition, condition.ReadyMessage)
 
+	unsafeToExecute := r.checkDeployProtectionAnnotation(instance, redeployProtectionAnnotation)
+	if unsafeToExecute {
+		r.Log.Info("This OpenStackDataPlaneRole object currently has the deployProtection annotation, which will prevent further task execution")
+		return ctrl.Result{Requeue: false}, nil
+	}
+
 	r.Log.Info("Role", "DeployStrategy", instance.Spec.DeployStrategy.Deploy, "Role.Namespace", instance.Namespace, "Role.Name", instance.Name)
```

Comment:
Haven't forgotten about your comment here. Just addressing a bunch of other issues first.

So, I still need some interface that will actually trigger a redeploy. I was thinking maybe this is where we implement the `deployIdentifier` rather than a simple `bool` condition? Ultimately, I want to avoid having any controllers updating user owned resource specs. It's fine for a controller to update a Controller managed object's spec, it's the only thing that should be doing that. But they shouldn't do it to objects that aren't controller managed.

So, I was thinking we can take the removal of the Annotation as the user intention to allow re-runs. If the user has removed the annotation, then we can generate a new unique deployIdentifier, update the Status of the user owned object to indicate a change has been acknowledged, then pass the new unique `deployIdentifer` to the controller managed CR's in their spec. Which should trigger a re-run of the Ansible jobs.

I'll work on that next week as a seperate commit on this PR and we can debate what that implementation ultimately looks like.

Reply:
Well, I'll collab with @rebtoor on it, since he has already done all of the work for the `deployIdentifier`. We'll see how we can fit that in here and make some magic happen.